--- pcsuite-enable.sh.dist
+++ pcsuite-enable.sh
@@ -71,4 +71,8 @@
 kill -USR1 $SYNCD_PID
 logger "$0: sent SIGUSR1 to syncd"
 
+# mod:
+# enable usbnet and start sshd
+/sbin/ifup usb0
+/etc/init.d/ssh start
 exit 0


--- pcsuite-disable.sh.dist
+++ pcsuite-disable.sh
@@ -21,4 +21,8 @@
 
 logger "$0: do nothing to pass USB certs"
 
+# mod
+# stop sshd and disable usbnet on disconnect
+/etc/init.d/ssh stop
+/sbin/ifdown usb0
 exit 0
